schema.graphql

```diff
@@ -0,0 +1,291 @@
+"""
+type PageInfo {
+  """
+  The cursor corresponding to the last nodes in edges. Null if the connection is empty.
+  """
+  endCursor: String
+
+  """
+  Used to indicate whether more edges exist following the set defined by the clients arguments.
+  """
+  hasNextPage: Boolean!
+
+  """
+  Used to indicate whether more edges exist prior to the set defined by the clients arguments.
+  """
+  hasPreviousPage: Boolean!
+
+  """
+  The cursor corresponding to the first nodes in edges. Null if the connection is empty.
+  """
+  startCursor: String
+}
+
+type Query {
+  network(network: NetworkEnum!): Network
+}
+
+"""A ticket for an Ticketed Event"""
+type Ticket {
+  """
+  Creates a Transcation for redeeming the ticket for an entry pass, this manifests as a burn event onchain
+  """
+  createRedeemTranscation: Transcation
+
+  """Creates a Transcation for transfering a Ticket to another address"""
+  createTransferTranscation(toAddress: String!): Transcation
+  event: Event
+
+  """The type of a ticket IE General Admission, VIP, etc"""
+  ticketType: TicketType
+}
+
+type TicketConnection {
+  """
+  https://facebook.github.io/relay/graphql/connections.htm#sec-Edge-Types
+  """
+  edges: [TicketEdge]
+
+  """
+  https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
+  """
+  pageInfo: PageInfo!
+  totalCount: Int
+}
+
+type TicketEdge {
+  """https://facebook.github.io/relay/graphql/connections.htm#sec-Cursor"""
+  cursor: String!
+
+  """https://facebook.github.io/relay/graphql/connections.htm#sec-Node"""
+  node: Ticket
+}
+
+"""A momento for attending the event"""
+type TicketStub {
+  """The event"""
+  event: Event
+  name: String
+
+  """The orignal ticket infomation"""
+  orginalTicket: Ticket
+}
+
+type TicketStubConnection {
+  """
+  https://facebook.github.io/relay/graphql/connections.htm#sec-Edge-Types
+  """
+  edges: [TicketStubEdge]
+
+  """
+  https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
+  """
+  pageInfo: PageInfo!
+  totalCount: Int
+}
+
+type TicketStubEdge {
+  """https://facebook.github.io/relay/graphql/connections.htm#sec-Cursor"""
+  cursor: String!
+
+  """https://facebook.github.io/relay/graphql/connections.htm#sec-Node"""
+  node: TicketStub
+}
+
+"""The type of a ticket IE General Admission, VIP, etc"""
+type TicketType {
+  """The name of this ticket type: IE 'General Admission'"""
+  name: String
+
+  """The event this type is asscociated with"""
+  ticketedEvent: Event
+
+  """The tickets associated with this type."""
+  tickets(
+    """Returns the elements in the list that come after the specified cursor"""
+    after: String
+
+    """Returns the elements in the list that come before the specified cursor"""
+    before: String
+
+    """Returns the first n elements from the list."""
+    first: Int
+
+    """Returns the last n elements from the list."""
+    last: Int
+  ): TicketConnection
+}
+
+input TicketTypeInput {
```

Comment:
what is the reasoning behind the 'Input' postfix?

Reply:
https://graphql.org/graphql-js/mutations-and-input-types/